src/PDPVerifier.sol

```diff
@@ -481,6 +477,11 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
     function nextProvingPeriod(uint256 setId, uint256 challengeEpoch, bytes calldata extraData) public {
         require(extraData.length <= EXTRA_DATA_MAX_SIZE, "Extra data too large");
         require(msg.sender == proofSetOwner[setId], "only the owner can move to next proving period");
+        
+        if (proofSetLastProvenEpoch[setId] == NO_PROVEN_EPOCH) {
+            proofSetLastProvenEpoch[setId] = block.number;
```

Comment:
@ZenGround0 I am not sure this is correct. This should only be updated in `provePossesion` after it is successful.
What's the rationale here ?

Reply:
Why not just initialise this the first time a root is added to a proof set and then only update it on successful calls to `provepossesion` ?

Reply:
Okay makes sense. 

Also, I was worried that this means that the prover now has to pay low proof fees even if they fault certain periods but realise now that we only update `proofSetLastProvenEpoch` here if it is 0. This means that this only applies to the transition of going from an proofset not being proven at all to it being proven and that is fine.